docs/source/model_doc/detr.rst

```diff
@@ -0,0 +1,117 @@
+naturally extended to perform panoptic segmentation, by simply adding a mask head on top of the decoder outputs.
+
+The abstract from the paper is the following:
+
+*We present a new method that views object detection as a direct set prediction problem. Our approach streamlines the
+detection pipeline, effectively removing the need for many hand-designed components like a non-maximum suppression
+procedure or anchor generation that explicitly encode our prior knowledge about the task. The main ingredients of the
+new framework, called DEtection TRansformer or DETR, are a set-based global loss that forces unique predictions via
+bipartite matching, and a transformer encoder-decoder architecture. Given a fixed small set of learned object queries,
+DETR reasons about the relations of the objects and the global image context to directly output the final set of
+predictions in parallel. The new model is conceptually simple and does not require a specialized library, unlike many
+other modern detectors. DETR demonstrates accuracy and run-time performance on par with the well-established and
+highly-optimized Faster RCNN baseline on the challenging COCO object detection dataset. Moreover, DETR can be easily
+generalized to produce panoptic segmentation in a unified manner. We show that it significantly outperforms competitive
+baselines.*
+
+The original code can be found `here <https://github.com/facebookresearch/detr>`__.
+
+Here's a TLDR explaining how the model works:
+
+First, an image is sent through a pre-trained convolutional backbone (in the paper, the authors use
+ResNet-50/ResNet-101). Let's assume we also add a batch dimension. This means that the input to the backbone is a
+tensor of shape :obj:`(1, 3, height, width)`, assuming the image has 3 color channels (RGB). The CNN backbone outputs a
+new lower-resolution feature map, typically of shape :obj:`(1, 2048, height/32, width/32)`. This is then projected to
+match the hidden dimension of the Transformer of DETR, which is :obj:`256` by default, using a :obj:`nn.Conv2D` layer.
+So now, we have a tensor of shape :obj:`(1, 256, height/32, width/32).` Next, the image is flattened and transposed to
+obtain a tensor of shape :obj:`(batch_size, seq_len, d_model)` = :obj:`(1, width/32*height/32, 256)`. So a difference
+with NLP models is that the sequence length is actually longer than usual, but with a smaller :obj:`d_model` (which in
+NLP is typically 768 or higher).
+
+Next, this is sent through the encoder, outputting :obj:`encoder_hidden_states` of the same shape (you can consider
+these as image features). Next, so-called **object queries** are sent through the decoder. This is a tensor of shape
+:obj:`(batch_size, num_queries, d_model)`, with :obj:`num_queries` typically set to 100 and is initialized with zeros.
+Each object query looks for a particular object in the image. Next, the decoder updates these object queries through
```

Comment:
When reading this, I understand "one type of object" (so one head looks for all balloons, another for all horses etc.) I think each head is looking for "one object" but two heads can look for the same type (if for instance there are two balloons/two horses) so I would precise:

Each object query looks for a particular object in the image (two different heads can be looking for the same object types if there are several objects of the same type in the image).

Reply:
Actually, re-reading the paper, I don't fully understand it: the input to the decoder is called tgt (target) and is initialized with zeros, and then these object queries are added at each attention layer. It's a bit confusing to me what the difference is between the input to the decoder and the query position embeddings.

Reply:
Update: [this issue](https://github.com/facebookresearch/detr/issues/225#issuecomment-688799449) clarifies it.